File: tests/BjjEire.Api.IntegrationTests/Support/ApiRoutes.cs
// Copyright (c) BjjEire. All rights reserved.
// Licensed under the MIT License.

namespace BjjEire.Api.IntegrationTests.Support;

internal static class ApiRoutes
{
    public const string ApiVersion = "v1";
    public const string ApiBase = $"/api/{ApiVersion}";

    public const string BjjEvents = $"{ApiBase}/bjjevent";
    public const string Competitions = $"{ApiBase}/competition";
    public const string FeatureFlags = $"{ApiBase}/featureflag";
    public const string Gyms = $"{ApiBase}/gym";
    public const string Stores = $"{ApiBase}/store";

    public const string OpenApiDocument = $"/openapi/{ApiVersion}.json";
}

File: src/bjjeire-app/src/config/api-routes.ts
export const API_VERSION = 'v1'
export const API_BASE_PATH = `/api/${API_VERSION}`

export const API_RESOURCE_ROUTES = {
  gyms: '/gym',
  bjjEvents: '/bjjevent',
  competitions: '/competition',
  stores: '/store',
  featureFlags: '/featureflag',
} as const

export const API_ROUTES = {
  gyms: `${API_BASE_PATH}${API_RESOURCE_ROUTES.gyms}`,
  bjjEvents: `${API_BASE_PATH}${API_RESOURCE_ROUTES.bjjEvents}`,
  competitions: `${API_BASE_PATH}${API_RESOURCE_ROUTES.competitions}`,
  stores: `${API_BASE_PATH}${API_RESOURCE_ROUTES.stores}`,
  featureFlags: `${API_BASE_PATH}${API_RESOURCE_ROUTES.featureFlags}`,
  donateBitcoinQr: `${API_BASE_PATH}/donate/bitcoin/qr`,
} as const
